Source/JobGiver_ZombieResponse.cs

```diff
@@ -13,87 +13,61 @@ public class JobGiver_ZombieResponse : ThinkNode_JobGiver
         // Token: 0x060003E6 RID: 998 RVA: 0x0002923C File Offset: 0x0002763C
         protected override Job TryGiveJob(Pawn pawn)
         {
-            return this.TryGetAttackNearbyEnemyJob(pawn);
+            Pawn_Zombiefied predator = pawn as Pawn_Zombiefied;
+            Pawn prey = BestPawnToHuntForPredator(predator);
+            if (prey == null) return null;
+            return new Job(ZombiefiedMod.zombieHunt, prey);
         }
 
-        // Token: 0x060003E7 RID: 999 RVA: 0x000292AC File Offset: 0x000276AC
-        private Job TryGetAttackNearbyEnemyJob(Pawn pawn)
-        {
-            Pawn thing = BestPawnToHuntForPredator(pawn);
-            if (thing != null)
-            {
-                return new Job(ZombiefiedMod.zombieHunt, thing)
-                {
-                    killIncappedTarget = true,
-                    expiryInterval = (int)(Rand.RangeSeeded(1f, 2f, Find.TickManager.TicksAbs) * 700),
-                    attackDoorIfTargetLost = true
-                };
-            }
-            return null;
-        }
 
-        public Pawn BestPawnToHuntForPredator(Pawn predator, float range = 7f)
+        public Pawn BestPawnToHuntForPredator(Pawn_Zombiefied predator, float range = 7f)
         {
-            //List<Pawn> allPawnsSpawned = predator.Map.mapPawns.AllPawnsSpawned;
-            List<Thing> allThingsRegion = predator.GetRegion().ListerThings.AllThings;
+            List<Thing> allThingsRegion = predator.Map.listerThings.AllThings;
```

Comment:
Checking the whole map is expensive. Is region not enough?

Reply:
You know what I did that to test, but it might not be enough if region limits it to just the room it's in, 🤔 Can't target a pawn through a wall, and makes the feature just not work. 

However, we can probably at the very least put this behind a check for the feature configuration. I'll test and update that.

Reply:
Ah so region refers to room? I didnt know. Just keep in mind that ZombieResponse gets called virtually always and for every single zombie so performance is important here and iirc using region instead of map increased performance by a lot

Reply:
It could, I feel like that's what I was running into when initially testing, but I could be wrong on that.

Reply:
Ah maybe I'm a bit wrong on that, so regions are kind of like rooms but more partitions of the map, the problem is they partition off where rooms intersect.  The problem with this, is I can't get pawns on the other side of a wall if this is the case.

https://www.dropbox.com/s/a1w6jaxty9hzobv/Screenshot%20%2891%29.png?dl=0

Was testing with this as the line to hopefully solve it using the `RegionType` parameter, but it doesn't seem to work here.
```cs
List<Thing> allThingsRegion = predator.GetRegion(RegionType.Set_All).ListerThings.ThingsMatching(ThingRequest.ForGroup(ThingRequestGroup.Pawn));```